aep/general/0231/aep.md.j2

```diff
@@ -1,5 +1,168 @@
+  names specifying the resources to retrieve. The field **should** be named
+  `names`.
+  - If no resource names are provided, the API **should** error with
+    `INVALID_ARGUMENT`.
+  - The field **should** be required.
+  - The field **should** identify the [resource type][aep-122-names] that it
+    references.
+  - The comment for the field **should** document the resource pattern.
+- Other fields besides `name` **may** be "hoisted" from the [standard Get
+  request][request-message]. There is no way to allow for these fields to
+  accept different values for different resources; if this is needed, use the
+  [alternative request form](#nested-request-objects).
+- Batch get **should not** support pagination because transactionality across
+  API calls would be extremely difficult to implement or enforce, and the
+  request defines the exact scope of the response anyway.
+- The request **must not** contain any other required fields, and **should
+  not** contain other optional fields except those described in this or another
+  AEP.
+- The comment above the `names` field **should** document the maximum number of
+  requests allowed.
+
+{% tab oas %}
+
+- The request **must** include a query parameter which accepts an array of
+  resource names specifying the resources to retrieve. The parameter **should**
+  be named `names`.
+  - If no resource names are provided, the API **should** error with status
+    code `400 Bad Request`.
+  - The parameter **should** be required.
+  - The parameter **should** identify the [resource type][aep-122-names] that
+    it references.
+  - The comment for the parameter **should** document the resource pattern.
+- Other parameters besides `name` **may** be "hoisted" from the [standard Get
+  request][request-message]. There is no way to allow for these parameters to
+  accept different values for different resources; if this is needed, use the
+  [alternative request form](#nested-request-objects).
+- The request **must not** contain any other required parameters, and **should
+  not** contain other optional parameters except those described in this or
+  another AEP.
+- The `names` parameter **should** document the maximum number of values
+  allowed in the array.
+
+{% endtabs %}
+
+- Batch get **should not** support pagination because transactionality across
+  API calls would be extremely difficult to implement or enforce, and the
+  request defines the exact scope of the response anyway.
+
+### Response
+
+The response for a batch get method **should** be specified with the following
+pattern:
+
+{% tab proto -%}
+
+```proto
+message BatchGetBooksResponse {
+  // Books requested.
+  repeated Book books = 1;
+}
+```
+
+- The response message **must** include one repeated field corresponding to the
+  resources being retrieved.
+
+{% tab oas %}
+
+- The response schema **must** be an array with each item containing one of the
+  requested resources.
```

Comment:
Is this structurally different from the response you'd get with a protobuf API?  It seems like it:  `{"books": [...]}` vs. `[...]`.

I know we probably can't get full JSON parity between OAS and transcoded gRPC, but nevertheless it seems unfortunate to diverge here unless there's a good reason to do so.

Reply:
It doesn't look like that change was actually made...?

Also, it's not just `List` - the `BatchGetBooksResponse` defined above follows the same pattern.

Reply:
I think these are all fixed ... in the OAS parts anyway. If the proto is wrong I will need someone to tell me how to fix it.